test/common/utxobased/engine/engine.spec.ts

```diff
@@ -0,0 +1,479 @@
+    })
+
+    // it('Add Future Addresses', function () {
+    //   await engine.addGapLimitAddresses(future)
+    // })
+  })
+
+  describe('Should start engine', function () {
+    it.skip('Get BlockHeight', function (done) {
+      const { uri, defaultHeight } = tests.BlockHeight
+      this.timeout(3000)
+      const testHeight = (): void => {
+        emitter.on('onBlockHeightChange', height => {
+          if (height >= heightExpected) {
+            emitter.removeAllListeners('onBlockHeightChange')
+            assert(engine.getBlockHeight() >= heightExpected, 'Block height')
+            done() // Can be "done" since the promise resolves before the event fires but just be on the safe side
+          }
+        })
+        engine.startEngine().catch(e => {
+          fakeLogger.info('startEngine error', e, e.message)
+        })
+      }
+      let heightExpected = defaultHeight
+      if (uri !== '') {
+        request.get(uri, (err, _res, body) => {
+          assert(err != null, 'getting block height from a second source')
+          const thirdPartyHeight = parseInt(JSON.parse(body).height)
+          if (thirdPartyHeight !== 0 && !isNaN(thirdPartyHeight)) {
+            heightExpected = thirdPartyHeight
+          }
+          testHeight()
+        })
+      } else {
+        testHeight()
+      }
+    })
+  })
+
+  describe(`Get Wallet Keys for Wallet type ${WALLET_TYPE}`, function () {
+    it('get private key', function (done) {
+      engine.getDisplayPrivateSeed()
+      done()
+    })
+    it('get public key', function (done) {
+      engine.getDisplayPublicSeed()
+      done()
+    })
+  })
+
+  // describe(`Is Address Used for Wallet type ${WALLET_TYPE} from network`, function () {
+  //   it('Checking a non empty P2WSH address', function (done) {
+  //     setTimeout(() => {
+  //       assert.equal(engine.isAddressUsed('tb1qzsqz3akrp8745gsrl45pa2370gculzwx4qcf5v'), true)
+  //       done()
+  //     }, 1000)
+  //   })
+
+  //   it('Checking a non empty address P2SH', function (done) {
+  //     setTimeout(() => {
+  //       assert.equal(engine.isAddressUsed('2MtegHVwZFy88UjdHU81wWiRkwDq5o8pWka'), true)
+  //       done()
+  //     }, 1000)
+  //   })
+  // })
+
+  /* TODO
+  describe(`Get Fresh Address for Wallet type ${WALLET_TYPE}`, function () {
+    it('Should provide a non used BTC address when no options are provided', async function () {
+      this.timeout(3000)
+      const address = await engine.getFreshAddress({}) // TODO
+      const processor = await makeProcessor({
+        disklet: engineOpts.walletLocalDisklet
+      })
+
+      const txs = await processor.fetchTransactions({ blockHeight: 0 })
+
+      // $FlowFixMe
+      const engineState: any = engine.engineState
+      const scriptHash = engineState.scriptHashes[address.publicAddress]
+      const transactions = engineState.addressInfos[scriptHash].txids
+      assert(transactions.length === 0, 'Should have never received coins')
+    })
+  })
+  */
```

Comment:
Figure out why it's commented out instead of just skipped.

Reply:
It was commented out because it was an attempt to rewrite the original test before realizing that the fixture data was unobtainable at the time.